source/Irrlicht/CMakeLists.txt

```diff
@@ -264,6 +264,16 @@ add_library(IrrlichtMt)
 foreach(object_lib
 	IRRMESHOBJ IRROBJ IRRVIDEOOBJ
 	IRRIOOBJ IRROTHEROBJ IRRGUIOBJ)
+	target_compile_options(${object_lib}
+		PRIVATE
+			$<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:
+				# The strlen optimization causes a stringop-overflow warning.
+				# The optimization caused no difference in the binary of
+				# a small reproducible case, suggesting that
+				# the warning is erroneous. However, I see no harm in
+				# disabling the optimization, just in case.
```

Comment:
I don't think disabling optimizations to get rid of mere warnings is harmless.

Reply:
You're right. I realized that solution was a lazy workaround and I've now fixed the warning properly, by improving the code quality.